core/classes/Group_Sync/GroupSyncManager.php

```diff
@@ -283,6 +287,129 @@ public function broadcastChange(User $user, string $sending_injector_class, arra
+                if ($injector == $sending_injector) {
+                    continue;
+                }
+
+                $injector_class = get_class($injector);
+
+                $batchable = $injector instanceof BatchableGroupSyncInjector;
+                if ($batchable && !array_key_exists($injector_class, $batched_changes)) {
+                    $batched_changes[$injector_class] = [
+                        'add' => [],
+                        'remove' => [],
+                    ];
+                }
+
+                $injector_column = $injector->getColumnName();
+                $injector_group_id = $rule->{$injector_column};
+                $sending_group_id = $rule->{$sending_injector->getColumnName()};
+
+                // Skip this injector if it doesn't have a group id setup for this rule
+                if ($injector_group_id === null) {
+                    continue;
+                }
+
+                if (!isset($modified[$injector_column])) {
+                    $modified[$injector_column] = [];
+                }
+
+                // Skip this specific injector for this rule if we have already modified the user
+                // with the same injector group id
+                if (in_array($injector_group_id, $modified[$injector_column])) {
+                    continue;
+                }
+
+                if (in_array($sending_group_id, $group_ids_add)) {
+                    // Add group to user
+                    $modified[$injector_column][] = $injector_group_id;
+                    if ($batchable) {
+                        $batched_changes[$injector_class]['add'][] = $injector_group_id;
+                    } elseif ($injector->addGroup($user, $injector_group_id)) {
+                        $logs['added'][] = "{$injector_column} -> {$injector_group_id}";
+                    }
+                } elseif (in_array($sending_group_id, $group_ids_remove)) {
+                    // Remove group from user
+                    $modified[$injector_column][] = $injector_group_id;
+                    if ($batchable) {
+                        $batched_changes[$injector_class]['remove'][] = $injector_group_id;
+                    } elseif ($injector->removeGroup($user, $injector_group_id)) {
+                        $logs['removed'][] = "{$injector_column} -> {$injector_group_id}";
+                    }
+                }
+            }
+        }
+
+        foreach ($batched_changes as $injector_class => $data) {
+            $add = $data['add'];
+            $remove = $data['remove'];
+
+            /** @var GroupSyncInjector&BatchableGroupSyncInjector $injector */
+            $injector = $this->getInjectorByClass($injector_class);
+            $injector_column = $injector->getColumnName();
+
+            if ($injector instanceof BatchableGroupSyncInjector) {
+                /** @var GroupSyncInjector&BatchableGroupSyncInjector $injector */
+                $batchable_injector = $injector;
```

Comment:
Unused assignment?

Reply:
Right, I was rewriting this code but forgot to finish it